src/libcollections/btree/node.rs

```diff
@@ -43,17 +53,39 @@ use boxed::Box;
 const B: usize = 6;
 pub const CAPACITY: usize = 2 * B - 1;
 
+/// The underlying representation of leaf nodes. Note that is often unsafe to actually store
+/// these, since only the first `len` keys and values are assumed to initialized. See also
+/// rust-lang/rfcs#197, which would make this structure significantly more safe by avoiding
+/// accidentally dropping unused and uninitialized keys and values.
 struct LeafNode<K, V> {
+    // The arrays storing the actual data of the node. Only the first `len` elements of each
+    // array are initialized and valid.
     keys: [K; CAPACITY],
     vals: [V; CAPACITY],
+
+    // We use `*const` as opposed to `*mut` so as to be covariant in `K` and `V`.
+    // This either points to an actual node or is null.
     parent: *const InternalNode<K, V>,
+
+    // The index into the parent node's `edges` array. `*node.parent.edges[node.parent_idx]`
+    // should be the same thing as `node`.
+    // This is only guaranteed to be initialized when `parent` is nonnull.
     parent_idx: u16,
+
+    // The number of keys and values this node stores.
+    // This is at the end of the node's representation and next to `parent_idx` to encourage
+    // the compiler to join `len` and `parent_idx` into the same 32-bit word, reducing space
+    // overhead.
```

Comment:
Why not make all these comment proper doc comments? May be useful if good idea support for working libstd ever happens.